src/PackageDocs/index.tsx

```diff
@@ -1,27 +1,140 @@
-import { Box } from "@chakra-ui/react";
-import ReactMarkdown from "@uiw/react-markdown-preview";
+import { LinkIcon } from "@chakra-ui/icons";
+import { Flex, Box, Heading, As } from "@chakra-ui/react";
 import ChakraUIRenderer from "chakra-ui-markdown-renderer";
 import * as reflect from "jsii-reflect";
+import React, { useState } from "react";
+import ReactMarkdown from "react-markdown";
 import { Documentation } from "../api/docgen/view/documentation";
+import { PackageNav } from "../PackageNav";
 
 export interface PackageDocsProps {
   assembly: reflect.Assembly;
   language: string;
   submodule?: string;
 }
 
+type HeadingResolverProps = {
+  level: number;
+  children: React.ReactNode;
+};
+
+const Headings =
+  (pushItems: (item: { title: string; path: string; level: number }) => void) =>
+  ({ level, children }: HeadingResolverProps) => {
+    // Access actual (string) value of heading
+    const size: string = ["2xl", "xl", "lg", "md", "sm", "xs"][level - 1];
+    const elem = `h${level}` as As<any>;
+    let title = "";
+    let anchor = "";
+    React.Children.map(children, (child) => {
+      if (typeof child === "string") {
+        title += child;
+        anchor += child.toLowerCase();
+      }
+    });
+
+    // Clean anchor (replace special characters whitespaces).
+    // Alternatively, use encodeURIComponent() if you don't care about
+    // pretty anchor links
+    anchor = anchor.replace(/[^a-zA-Z0-9 ]/g, "");
+    anchor = anchor.replace(/ /g, "-");
+    pushItems({ level, path: anchor, title });
+
+    return (
+      <Heading my={4} level={level} as={elem} size={size}>
+        <a id={anchor} href={`#${anchor}`}>
+          <LinkIcon w={4} h={4} color="gray.500" mr={2} />
+        </a>
+        {children}
+      </Heading>
+    );
+  };
+
+interface NavItem {
+  title: string;
+  path: string;
+  children: NavItem[];
+}
 export function PackageDocs(props: PackageDocsProps) {
+  const [navItems, setNavItems] = useState<NavItem[]>([]);
   const doc = new Documentation({
```

Comment:
I don't know how expensive it is to create an instance of `Documentation`, but assuming that it is, we will probably want to memoize this so that it is not created on every render i.e when `props` change or `navItems` state changes.

I suggest using `useMemo`

```ts
const { assembly, language, submodule } = props; // or destructure in parameters

const doc = useMemo(() => new Documentation({
  assembly,
  language,
  submoduleName: submodule,
}), [assembly, language, submodule]);
```

This will allow us to only need to redefine doc when any of those dependencies are changed